### ISSUE TITLE
Support fallback nicks when nick is already in use

### ISSUE THREAD

**Issue by SoniEx2** [thelounge/thelounge#4816]
### Feature Description

It would be nice to be able to specify fallback nicks, instead of adding random digits at the end.

**Comment by mokahless** [comment]
Not sure if it's related enough or if I should start a new issue for this but:

I think related, I'd like to be able to specify no fallback. Sometimes, I will return to thelounge months later to find my username permanently changed in settings from "username" to "username8" etc. And obviously because of this, the servers with proper username lookups and logins just stop working.

**Comment by Beersteddy** [comment]
> Not sure if it's related enough or if I should start a new issue for this but:
> 
> I think related, I'd like to be able to specify no fallback. Sometimes, I will return to thelounge months later to find my username permanently changed in settings from "username" to "username8" etc. And obviously because of this, the servers with proper username lookups and logins just stop working.

Same issue still exists which makes theLounge totally useless.

Can we just have option:
"Use set nickname, never use any other, especially random"

**Comment by jessielw** [comment]
> > Not sure if it's related enough or if I should start a new issue for this but:
> > 
> > I think related, I'd like to be able to specify no fallback. Sometimes, I will return to thelounge months later to find my username permanently changed in settings from "username" to "username8" etc. And obviously because of this, the servers with proper username lookups and logins just stop working.
> 
> Same issue still exists which makes theLounge totally useless.
> 
> Can we just have option:
> "Use set nickname, never use any other, especially random"
> 

+1 to this

**Comment by ppfeister** [comment]
@mokahless @btTeddy Did either of you ever stumble across a fix for this in the meantime?

The fact that it insists on overriding your very explicit configuration is **bad** -- so so so many issues due to that design choice

**Comment by jessielw** [comment]
Completely agree.

**Comment by brunnre8** [comment]
folks, me too comments aren't helpful.
The issue is open, what's needed is someone with enough skin in the game to fix it.

If you don't plan on working on it, just move along

**Comment by jessielw** [comment]
> folks, me too comments aren't helpful.
> The issue is open, what's needed is someone with enough skin in the game to fix it.
> 
> If you don't plan on working on it, just move along 

This is true, my apologies. 

I'll try to find time to do a PR later. But the current implementation would have taken more time and effort to implement than a simple fixed username.

**Comment by rand0mdud3** [comment]
Isn't this already what #3404 is supposed to take care of ?

**Comment by cartfisk** [comment]
I also went down this rabbit-hole today- https://github.com/thelounge/thelounge/pull/3404 doesn't appear to be documented and is not surfaced in the client "Network" config page. I dug through to find the post-TS-rewrite code for that PR and it appears the feature _should_ still work by adding `"keepNick": true,` to network objects in a `users/user.json` config file.

I'll wait a few days to see if the feature works as expected and then look into adding a checkbox to the client or at least documenting the JSON config flag.

**Comment by jessielw** [comment]
> I also went down this rabbit-hole today- https://github.com/thelounge/thelounge/pull/3404 doesn't appear to be documented and is not surfaced as a control in the client "Network" config page UI. I dug through to find the post-TS-rewrite code for that PR and it appears the feature _should_ still work by adding `"keepNick": true,` to network objects in a `users/user.json` config file.
> 
> I'll wait a few days to see if the feature works as expected and then look into adding a checkbox to the client or at least documenting the JSON config flag.

Let me know your findings!

**Comment by TrueGreatMerge** [comment]
If I add `"keepNick": true` in a `users/user.json` it's just removed after a few seconds to a minute.

**Comment by dgw** [comment]
@TrueGreatMerge Editing the file while The Lounge is running? If so, don't. Stop the program first.

**Comment by TrueGreatMerge** [comment]
> [@TrueGreatMerge](https://github.com/TrueGreatMerge) Editing the file while The Lounge is running? If so, don't. Stop the program first.

Thanks for reply. I am running docker compose with :latest tag. 

1. I exited the container.
2. Edited the `user.json` in the users folder and put `"keepNick": true,` in the line under each `"nick": "abc",` in the `networks: [ {` .
3. Saved `user.json` file.
4. Pulled latest the lounge docker version.
5. Started the container.
6. The `"keepNick": true,` all disappeared.

**Comment by jessielw** [comment]
> > [@TrueGreatMerge](https://github.com/TrueGreatMerge) Editing the file while The Lounge is running? If so, don't. Stop the program first.
> 
> Thanks for reply. I am running docker compose with :latest tag.
> 
>     1. I exited the container.
> 
>     2. Edited the `user.json` in the users folder and put `"keepNick": true,` in the line under each `"nick": "abc",` in the `networks: [ {` .
> 
>     3. Saved `user.json` file.
> 
>     4. Pulled latest the lounge docker version.
> 
>     5. Started the container.
> 
>     6. The `"keepNick": true,` all disappeared.

I took a look at the code. It doesn't work because it doesn't exist in the code base.

**Comment by jessielw** [comment]
> > Not sure if it's related enough or if I should start a new issue for this but:
> > I think related, I'd like to be able to specify no fallback. Sometimes, I will return to thelounge months later to find my username permanently changed in settings from "username" to "username8" etc. And obviously because of this, the servers with proper username lookups and logins just stop working.
> 
> Same issue still exists which makes theLounge totally useless.
> 
> Can we just have option: "Use set nickname, never use any other, especially random"

Did a PR to sort this out. With https://github.com/thelounge/thelounge/pull/4973 it'll simply wait for your nick to become available and not append random digits. Waiting on the maintainers to take a look at it and see what they think.

**Comment by blixten85** [comment]
It would be handy if it could re-run the commands in the command list on each server when it regains the standard nick. 

However, it changes the original nick in the server configuration (still?) with a random number at the end, I’m not sure if a fix has been published about keepnick true or not?
